### PR TITLE
implemented Spring Boot JPA hw

### PR DESCRIPTION
**Pull request by ynik404**
Replaced the existing BookRepository with the one that will use JpaRepository interface
Added Liquibase support
Changed spring.jpa.hibernate.ddl-auto property value to validate

### CODE REVIEW

File: src/main/java/mate/academy/model/Book.java [hunk 1]
```diff
@@ -9,10 +9,14 @@
...
 import lombok.Getter;
 import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Getter
 @Setter
 @Entity
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/Book.java:19]: `@Where` is deprecated. Use `@SQLRestriction` instead

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Please, add .idea, *.iml, target/
!.mvn/wrapper/maven-wrapper.jar
!**/src/main/**/target/
!**/src/test/**/target/

to gitignore file

File: .idea/.gitignore
Comment (kerrrusha) [inline .idea/.gitignore:1]: idea folder should not be committed. remove it from git

File: hibernate-relations-hw.iml
Comment (kerrrusha) [inline hibernate-relations-hw.iml:1]: same

File: target/checkstyle-cachefile
Comment (kerrrusha) [inline target/checkstyle-cachefile:1]: same about target folder

File: src/main/java/mate/academy/model/Book.java [hunk 2]
```diff
@@ -29,4 +33,7 @@ public class Book {
     private BigDecimal price;
     private String description;
     private String coverImage;
+    @Column(nullable = false)
+    @SQLRestriction("is_deleted = false")
```
Comment (kerrrusha) [inline src/main/java/mate/academy/model/Book.java:37]:
```suggestion
```

**Review by kerrrusha — CHANGES_REQUESTED** [review]

**Pull request closed without merging**